docs/bugs_and_glitches.md

```diff
@@ -1,6 +1,43 @@
 # Bugs and Glitches
 
 
+## Contents
+
+- [Thick Club and Light Ball can decrease damage done with boosted (Special) Attack](#thick-club-and-light-ball-can-decrease-damage-done-with-boosted-special-attack)
+- [Metal Powder can increase damage taken with boosted (Special) Defense](#metal-powder-can-increase-damage-taken-with-boosted-special-defense)
+- [Belly Drum sharply boosts Attack even with under 50% HP](#belly-drum-sharply-boosts-attack-even-with-under-50-hp)
+- [HP bar animation is slow for high HP](#hp-bar-animation-is-slow-for-high-hp)
+- [HP bar animation off-by-one error for low HP](#hp-bar-animation-off-by-one-error-for-low-hp)
+- [Experience underflow for level 1 Pokémon with Medium-Slow growth rate](#experience-underflow-for-level-1-pokémon-with-medium-slow-growth-rate)
+- [Five-digit experience gain is printed incorrectly](#five-digit-experience-gain-is-printed-incorrectly)
+- [NPC use of Full Heal or Full Restore does not cure Nightmare status](#npc-use-of-full-heal-or-full-restore-does-not-cure-nightmare-status)
+- ["Smart" AI encourages Mean Look if its own Pokémon is badly poisoned](#smart-ai-encourages-mean-look-if-its-own-pokémon-is-badly-poisoned)
+- [A Disabled, PP Up–enhanced move may not trigger automatic Struggling](#a-disabled-pp-upenhanced-move-may-not-trigger-automatic-struggling)
+- [Counter and Mirror Coat still work if the opponent uses an item](#counter-and-mirror-coat-still-work-if-the-opponent-uses-an-item)
+- [Present damage is incorrect in link battles](#present-damage-is-incorrect-in-link-battles)
+- [BRN/PSN/PAR do not affect catch rate](#brnpsnpar-do-not-affect-catch-rate)
+- [Moon Ball does not boost catch rate](#moon-ball-does-not-boost-catch-rate)
+- [Love Ball boosts catch rate for the wrong gender](#love-ball-boosts-catch-rate-for-the-wrong-gender)
+- [Fast Ball only boosts catch rate for three Pokémon](#fast-ball-only-boosts-catch-rate-for-three-pokémon)
+- [Friend Ball catches sent to the PC overwrite the wrong Pokémon's happiness](#friend-ball-catches-sent-to-the-pc-overwrite-the-wrong-pokémons-happiness)
```

Comment:
This is actually not a bug. The ``SetBoxMonCaughtData`` at https://github.com/pret/pokecrystal/blob/master/items/item_effects.asm#L623 happens before the box Pokemon happiness is written to.

Reply:
Interesting! This GlitchCity thread agrees: <https://forums.glitchcity.info/index.php?topic=8105.0> I do wonder why the comment has been there all this time though.

Reply:
While we're at it, I'm lazy to properly submit a PR, but I know of two more battle bugs:
- Beat Up error, whose cause is located [here](https://github.com/pret/pokecrystal/blob/d0d61cb/battle/effect_commands.asm#L3194) (should be cp c), and a more detailed explanation can be found [here](https://www.youtube.com/watch?v=202-iAsrIa).
- [Substitute + Defense Down side effect error](https://www.youtube.com/watch?v=OGwKPRJLaaI) (see video description and https://github.com/pret/pokecrystal/blob/fd14253/battle/moves/move_effects.asm#L556-L577; the second ``effectchance`` shouldn't be there)

Reply:
Thanks! The Beat Up video is unavailable; is it private, or the link wrong?